File: admin/src/types/shopByCategory.ts
export type LinkType = 'category' | 'subcategory' | 'product_type' | 'product' | 'custom_url';

export interface ShopByCategoryItem {
  id: string;
  internal_name: string;
  title_en: string;
  title_ar: string;
  image_url: string;
  alt_text: string;
  link_type: LinkType;
  category_id: string;
  subcategory_id: string;
  product_type_id: string;
  product_id: string;
  custom_url: string;
  sort_order: number;
  is_active: boolean;
  is_published: boolean;
  created_at: string;
  updated_at: string;
}

export const INITIAL_SHOP_BY_CATEGORY: Omit<ShopByCategoryItem, 'id' | 'created_at' | 'updated_at'> = {
  internal_name: '',
  title_en: '',
  title_ar: '',
  image_url: '',
  alt_text: '',
  link_type: 'category',
  category_id: '',
  subcategory_id: '',
  product_type_id: '',
  product_id: '',
  custom_url: '',
  sort_order: 1,
  is_active: true,
  is_published: false,
};

File: backend/Mayar.Api/Entities/ShopByCategory.cs
using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace Mayar.Api.Entities;

public class ShopByCategory
{
    public Guid Id { get; set; }

    // Internal name for admin identification
    public string InternalName { get; set; } = string.Empty;

    // Bilingual titles
    public string TitleEnglish { get; set; } = string.Empty;
    public string? TitleArabic { get; set; }

    // Image
    public string ImageUrl { get; set; } = string.Empty;
    public string? AltText { get; set; }

    // Link targeting
    public string LinkType { get; set; } = "category"; // category, subcategory, product_type, product, custom_url
    public Guid? CategoryId { get; set; }
    public Guid? SubcategoryId { get; set; }
    public Guid? ProductTypeId { get; set; }
    public Guid? ProductId { get; set; }
    public string? CustomUrl { get; set; }

    // Navigation properties for related entities
    public TopCategory? Category { get; set; }
    public MiddleCategory? Subcategory { get; set; }
    public BottomCategory? ProductType { get; set; }
    public Product? Product { get; set; }

    // Admin controls
    public int SortOrder { get; set; } = 0;
    public bool IsActive { get; set; } = true;
    public bool IsPublished { get; set; } = false;

    // Timestamps
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}
